### PR TITLE
[ISSUE #3213] Allow to reset gas details

### LINKED ISSUE #3213: Reset to default and Done buttons are missing on Gas settings screen
### Description

*Type*: Bug

*Summary*: `Reset to default` and `Done` buttons are not present now on Gas settings screen. 

#### Expected behavior

Please check https://app.zeplin.io/project/5a55d587ba304eed4d2c942f/screen/5a5cfee281c5d84b3e8ad58b for UI specification

image

#### Actual behavior

image

### Reproduction

- Open Status
- Tap Wallet
- Tap Send -> Transaction Fee

### Additional Information
* Operating System: Android, iOS

### PR DESCRIPTION
**Pull request by jeluard**
fixes #3213

### Summary:

Allow to reset gas details to default value

### Steps to test:
- Open Status
- Change wallet fee in `send` screen
- Validate reset button works as expected

status: ready

### CODE REVIEW

File: src/status_im/ui/screens/db.cljs [hunk 1]
```diff
@@ -18,11 +18,14 @@
...
-(def transaction-send-default
-  {:symbol    :ETH
-   :gas       ethereum/default-transaction-gas
+(defn gas-default []
+  {:gas       ethereum/default-transaction-gas
    :gas-price ethereum/default-gas-price})
 
+(defn transaction-send-default []
+  (merge (gas-default)
+         {:symbol    :ETH}))
```
Comment (alwx) [inline src/status_im/ui/screens/db.cljs]: why is it a function? seems like it can be easily transformed to variable.

**Review by alwx — CHANGES_REQUESTED** [review]

File: src/status_im/ui/screens/db.cljs [hunk 2]
```diff
@@ -42,7 +45,7 @@
              :tags                       []
              :sync-state                 :done
              :wallet.transactions        constants/default-wallet-transactions
-             :wallet                     {:send-transaction transaction-send-default}
+             :wallet                     {:send-transaction (transaction-send-default)}
```
Comment (flexsurfer) [inline src/status_im/ui/screens/db.cljs]: why it became a function?

**Review by flexsurfer — APPROVED** [review]

**Review by goranjovic — APPROVED** [review]

File: src/status_im/ui/screens/db.cljs (hunk 1 above)
Comment (jeluard) [reply src/status_im/ui/screens/db.cljs]: I did this change to prepare for the move to a computed gas price (#3241)

File: src/status_im/ui/screens/db.cljs (hunk 2 above)
Comment (jeluard) [reply src/status_im/ui/screens/db.cljs]: I did this change to prepare for the move to a computed gas price (#3241)

File: src/status_im/ui/screens/db.cljs (hunk 1 above)
Comment (flexsurfer) [reply src/status_im/ui/screens/db.cljs]: could you leave a comment?

**Comment by jeluard** [comment]
@alwx @flexsurfer Reverted to `def`

**Review by alwx — APPROVED** [review]

**Comment by annadanchenko** [comment]
Issue 1: Recipient field content is cleared on Send Transaction screen after using Reset to Default on Transaction Fee screen. I would expect that it resets only values related to this screen and keeps other value untouched. 

Found on iOS TF (from 00:19) https://app.testfairy.com/projects/4803590-status/builds/7671104/sessions/4/?accessToken=/dVbNaVMuWDVKuGWNm9HzN/it4g

Steps:
- Open Wallet
- Tap on Send transaction
- Choose Recipient as Jarrad from Recent Recipient
- Tap on Transaction Fee field
- Tap on Reset to Default button
- Tap on Done or on "x" to return to Send transaction screen
- Check content of Choose Recipient field. Expected: it contains Jarrad as recipient

Questions:
1. What is the difference between "x" and "Done" on Transaction Fee screen? Now both are closing the screen. Why to have both?
2. If STT was selected as Asset on Send transaction screen then is it expected that Reset to Default resets defaults for ETH not STT? For example, I've selected STT and by default Gas Limit is 105000. If tap on Reset to Default then defaults for ETH are used, so Gas Limit becomes 21 000 and if return to Send Transaction screen then Asset is ETH instead of selected STT. My own expectation: becaue button Reset to Defaults resides on Transaction Fee screen it should affect only fields that can be edited on this screen. So, only Gas Limit and Gwei. If user selected STT or specified amount/recipient on other screen then these values should remain untouched.
3. Do we expect to clean the content of Amount field on Send Transaction screen if Reset to Defaults is used On Transaction Fee screen where Amount is read-only? Should it be reset to empty or we should keep the value?

**Comment by jeluard** [comment]
@annadanchenko I will fix your points #2 and #3 as they sound very similar to the first issue you are describing.
Question #1 is open to discussion.

**Comment by denis-sharypin** [comment]
@annadanchenko, @jeluard X button works more like Back button, it doesn't save changed parameters on the screen only Done button does.

**Comment by jeluard** [comment]
@annadanchenko @denis-sharypin Made the change related to back button.

**Pull request merged**